src/wp-includes/class-wp-theme-json.php

```diff
@@ -2035,7 +2056,9 @@ public function get_styles_for_block( $block_metadata ) {
 					// the feature selector. This may occur when multiple block
 					// support features use the same custom selector.
 					if ( isset( $feature_declarations[ $feature_selector ] ) ) {
-						$feature_declarations[ $feature_selector ] = array_merge( $feature_declarations[ $feature_selector ], $new_feature_declarations );
+						foreach ( $new_feature_declarations as $new_feature_declaration ) {
+							$feature_declarations[ $feature_selector ][] = $feature_declaration;
```

Comment:
@aristath @desroj @felixarntz @spacedmonkey @jrfnl @hellofromtonya 

Shouldn't the variable assigned be `$new_feature_declaration` instead of `$feature_declaration`? This has been reported as problematic by the Gutenberg lint jobs ([see conversation](https://github.com/WordPress/gutenberg/pull/46579/files#r1049600844)).

I haven't looked at what regression this introduced but it'd be good to create a follow-up PR fixing it and adding a test case.

Reply:
@oandregal I concur. This is a bug which should be fixed.
```suggestion
							$feature_declarations[ $feature_selector ][] = $new_feature_declaration;
```

Reply:
🤦 yeah, looks like this was a typo.. Good catch 👍

Reply:
Happens to the best of us ;-)

Reply:
I'm not familiar with this code path, so asked folks to provide details for testing at https://github.com/WordPress/gutenberg/pull/46579#discussion_r1049600844 so we can add a new unit test and gauge how severe this is.

Reply:
AFAICS, this should go in the next 6.1 patch release.